src/main/java/org/folio/search/utils/CallNumberUtils.java

```diff
@@ -0,0 +1,182 @@
+   * @return true if character is supported, false - otherwise
+   */
+  public static boolean isSupportedCharacter(char character) {
+    return VALID_CHARACTERS_MAP.containsKey(character);
+  }
+
+  /**
+   * Provides integer value representation for the given character.
+   *
+   * @param character    - character to analyze.
+   * @param defaultValue - default value if character is not supported
+   * @return integer representation for char value
+   */
+  public static int getIntValue(char character, int defaultValue) {
+    return VALID_CHARACTERS_MAP.getOrDefault(character, defaultValue);
+  }
+
+  /**
+   * Converts incoming call-number into long value.
+   *
+   * <p>
+   * This algorithm takes first 10 character from call-number and converts them into long value using following
+   * approach:
+   * <ul>
+   *   <li>Each character has own unique int value ({@code ' '=0}, {@code '.'=6},
+   *   {@code '/'=7}, {@code '0'=8}, {@code '9'-17}, {@code 'A'=23}, {@code 'Z'=48})</li>
+   *   <li>each char numeric value multiplied by 52<sup>(10-{charPosition})</sup> (
+   *   52 - is the maximum base to not exceed long max value - 2<sup>63</sup>-1)</li>
+   *   <li>all received values are summed to the result value</li>
+   * </ul>
+   * </p>
+   *
+   * @param callNumber - effective shelving order value from query or from instance item to process
+   * @return numeric representation of given call-number value
+   */
+  public static Long getCallNumberAsLong(String callNumber) {
+    return callNumberToLong(callNumber, 0L, CN_MAX_CHARS);
+  }
+
+  /**
+   * Converts incoming call-number with considering first position into long value.
+   *
+   * <p>
+   * This algorithm takes first 10 character from call-number and converts them into long value using following
+   * approach:
+   * <ul>
+   *   <li>Each character has own unique int value ({@code ' '=0}, {@code '.'=6},
+   *   {@code '/'=7}, {@code '0'=8}, {@code '9'-17}, {@code 'A'=23}, {@code 'Z'=48})</li>
+   *   <li>each char numeric value multiplied by 52<sup>(10-{charPosition})</sup> (
+   *   52 - is the maximum base to not exceed long max value - 2<sup>63</sup>-1)</li>
+   *   <li>all received values are summed to the result value</li>
+   * </ul>
+   * </p>
+   *
+   * @param callNumber    - effective shelving order value from query or from instance item to process
+   * @param firstPosition - int representation for first position
+   * @return numeric representation of given call-number value
+   */
+  public static Long getCallNumberAsLong(String callNumber, int firstPosition) {
+    long startVal = convertChar(firstPosition, CN_MAX_CHARS);
+    return callNumberToLong(callNumber, startVal, CN_MAX_CHARS - 1);
+  }
+
+  private static long callNumberToLong(String callNumber, long startVal, int maxChars) {
+    var cleanCallNumber = cleanCallNumber(callNumber, maxChars);
+    if (StringUtils.isBlank(cleanCallNumber)) {
+      return 0L;
+    }
+    long result = startVal;
```

Comment:
![image](https://user-images.githubusercontent.com/94473337/235745048-c6bfaa05-014c-4085-8e75-72f6f51b5006.png)

Reply:
Check line 140 in the file, it does what described